src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -243,35 +207,35 @@ public void Dispose()
 
             this.serverCancel.Cancel();
 
-            this.logger.LogTrace("Stopping network peer server.");
-
-            try
-            {
-                this.ConnectedNetworkPeers.DisconnectAll("Node shutdown");
-            }
-            catch (Exception e)
-            {
-                this.logger.LogTrace("Exception occurred: {0}", e.ToString());
-            }
-
             this.logger.LogTrace("Stopping TCP listener.");
-            this.tcpListener?.Stop();
+            this.tcpListener.Stop();
 
             this.logger.LogTrace("Waiting for accepting task to complete.");
             this.acceptTask.Wait();
 
-            ICollection<NetworkPeerConnection> connections = this.connectionsById.Values;
-            if (connections.Count > 0)
+            try
             {
-                this.logger.LogInformation("Waiting for {0} connected clients to finish.", connections.Count);
-                foreach (NetworkPeerConnection connection in connections)
+                if (this.peersById.Count > 0)
```

Comment:
Why not do as before and use `if (peers .Count > 0)`

Reply:
Good point. Will fix

Reply:
Done